internal/suggestions/suggestions.go

```diff
@@ -116,44 +126,112 @@ func (s *Suggestions) CommitSuggestion(newFile []byte) error {
 	return nil
 }
 
-func (s *Suggestions) FindSuggestion(err error, previousSuggestionContext *string) (*Suggestion, error) {
-	errString := err.Error()
-	vErr := errors.GetValidationErr(err)
+type ErrorAndSuggestion struct {
+	errorNum   int
+	suggestion *Suggestion
+}
+
+// FindSuggestions returns one suggestion per given error, in order
+func (s *Suggestions) FindSuggestions(errs []error) ([]*Suggestion, error) {
+	suggestions := make([]*Suggestion, len(errs))
+	ch := make(chan ErrorAndSuggestion)
+	var wg sync.WaitGroup
+
+	for i, err := range errs {
+		wg.Add(1)
+		go s.FindSuggestionAsync(err, i, ch, &wg)
+	}
+
+	// close the channel in the background
+	go func() {
+		wg.Wait()
+		close(ch)
+	}()
+	// read from channel as they come in until its closed
+	for res := range ch {
+		suggestions[res.errorNum] = res.suggestion
```

Comment:
perhaps my lack of go skill is showing, but what happens with the last error suggestion -- does it successfully get recorded here?

Just wondering if there's any concurrency issues because I'd imagine as soon as `wg.Done` gets called for that, both `close(ch)` in the other goroutine and this line in the for loop would execute, right? So I'm imagining there's a chance it can sometimes get dropped?

Reply:
Im definitely not an expert in goroutines by any stretch but my understanding is that this is a standard pattern (I got this from somewhere online)

Reply:
I think it should be fine because `ch <-` executes before `wg.Done`, so ideally the main thread should be able to pull of the channel before it gets closed. So yeah this is probably fine and we can revisit if we notice anything weird about last error